.github/workflows/ci.yml

```
@@ -3,47 +3,100 @@ name: CI
 on:
   push:
     branches: [main]
+    tags: ['*']
+
   pull_request:
     branches: [main]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    name: Build [Node.js ${{ matrix.node-version }}]
+    runs-on: ubuntu-22.04
 
     strategy:
       matrix:
-        node-version: [18.x, 20.x, 22.x]
+        include:
+          - node-version: 18.x
+            publish: false
+          - node-version: 20.x
+            publish: true # Publish on npm
+          - node-version: 22.x
+            publish: false
 
     env:
       CI: true
 
     steps:
-      - uses: actions/checkout@v4
-      - name: Use Node.js ${{ matrix.node-version }}
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      ##########################################################################
+      # Build
+      ##########################################################################
+
+      - name: Set up Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
-      - run: corepack enable
-      - run: yarn install
-      - run: yarn build
-      - run: yarn test
 
-  lint:
-    runs-on: ubuntu-latest
+      - name: Enable Corepack
+        run: corepack enable
 
-    strategy:
-      matrix:
-        node-version: [22.x]
+      - name: Install
+        run: yarn install
 
-    env:
-      CI: true
+      - name: Lint
+        run: yarn lint
 
-    steps:
-      - uses: actions/checkout@v4
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v4
+      - name: Build
+        run: yarn build
+
+      - name: Test
+        run: yarn test
+
+      ##########################################################################
+      # Publish
+      ##########################################################################
+
+      - name: Check tag format
+        id: check-tag-format
+        if: startsWith(github.ref, 'refs/tags/') && matrix.publish
+        uses: nowsprinting/check-version-format-action@v4
+
+      - name: Exit on invalid tag format
+        if: startsWith(github.ref, 'refs/tags/') && !steps.check-tag-format.outputs.is_valid && matrix.publish
+        run: echo "Tag must follow SemVer convention. Aborting." && exit 1
+
+      - name: Get release type
+        if: startsWith(github.ref, 'refs/tags/') && matrix.publish
+        id: get-release-type
+        uses: actions/github-script@v7
         with:
-          node-version: ${{ matrix.node-version }}
-      - run: corepack enable
-      - run: yarn install
-      - run: yarn lint
+          script: |
+            const regex = /(alpha|beta)/
+            const refName = context.ref.replace('refs/tags/', '')
+            console.log(`Ref tag: ${refName}`)
+            const releaseTypeMatch = refName.match(regex)
+            if (!releaseTypeMatch) {
```

Untested, but I think it would be possible to get this information from `check-version-format-action`, like...

```suggestion
            if (steps.check-tag-format.outputs.stable) {
```

... and similarly to get the prerelease tag from `steps.check-tag-format.outputs.prerelease`

---

I've checked that and it's only marked as stable in versions `>= 1.0.0`, for instance, `0.0.33` is marked as not stable

---

I see. Ok, perhaps after we reach 1.0.0 then.